### PR TITLE
Pin version to grpcio in build-requirements.txt

### PR DESCRIPTION
**Pull request by AnandInguva**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/python/build-requirements.txt [hunk 1]
```diff
@@ -21,6 +21,9 @@ setuptools
 wheel>=0.36.0
 
 grpcio-tools==1.37.0
+# TODO(https://github.com/apache/beam/issues/23734): the sdist for grpcio==1.50.0 is failing on GH workers
+# pin grpcio to the previous version.
+grpcio==1.49.1
```
Comment (Abacn) [inline sdks/python/build-requirements.txt:26]: Should we pick the same version for grpcio and grpcio-tools? (We should probably do it at the beginning...)

Comment (AnandInguva) [reply sdks/python/build-requirements.txt:26]: What would be the consequence if the version between the both differ? 

IIUC, grpcio-tools==1.37.0 has a requirement of grpcio>=1.37.0. I think having a version equal to or greater than grpcio-tools for grpcio would be fine.

**Comment by AnandInguva** [comment]
R: @tvalentyn @TheNeuralBit

File: sdks/python/build-requirements.txt (hunk 1 above)
Comment (tvalentyn) [reply sdks/python/build-requirements.txt:26]: can we do this pinning for the mac platform only? Thanks!

Comment (AnandInguva) [reply sdks/python/build-requirements.txt]: Done

**Comment by tvalentyn** [comment]
LGTM

**Comment by tvalentyn** [comment]
remaining tests should be not affected by this change.

**Pull request merged**